Dockerfile.ci

```diff
@@ -0,0 +1,41 @@
+FROM registry.access.redhat.com/ubi8/go-toolset:1.17.7 
+
+USER root
+RUN mkdir -p /root/go/src/github.com/Azure/ARO-RP 
+WORKDIR /root/go/src/github.com/Azure/ARO-RP 
+ENV GOPATH=/root/go
+
+#we have multiple steps for copy so we can make use of caching
+COPY vendor/ vendor
+
+COPY docs/ docs
+COPY hack/ hack
+
+COPY swagger/ swagger
+COPY test/ test
+COPY python/ python
+COPY portal/ portal
+
+COPY cmd/ cmd
+COPY pkg/ pkg
+
+#COPY all files with an extension (directories not copied)
+COPY ?*.* .
+COPY Makefile LICENSE .
+
+COPY .git .git
+COPY .gitignore .gitignore
+COPY .pipelines .pipelines
+COPY .gdn .gdn
+COPY .github .github
+COPY .env .env
+COPY .sha256sum .sha256sum
+COPY .config .config
+
+RUN make generate
+RUN hack/git-utils/isClean.sh
+RUN make build-all
+RUN hack/git-utils/isClean.sh
+RUN make unit-test-go
+RUN hack/git-utils/isClean.sh
```

Comment:
nit: I wonder if putting this into it's own hack script, with proper error messaging, would be helpful. Multiple RUN clauses reduces performance. It might be helpful to have a single script that we can run everywhere, so we know our pipelines are the same.